tests/src/Unit/SocialApiUserTest.php

```diff
@@ -0,0 +1,182 @@
+    $messenger = $this->createMock(MessengerInterface::class);
+    $user_manager = $this->createMock(UserManagerInterface::class);
+
+    $this->dataHandler = $this->getMockBuilder(SocialApiDataHandler::class)
+      ->disableOriginalConstructor()
+      ->setMethods(['get', 'set', 'getSessionPrefix'])
+      ->getMock();
+
+    $entity_type = 'users';
+
+    $this->sessionKeys = [];
+
+    $this->userManager = $this->getMockBuilder(UserManager::class)
+      ->setConstructorArgs([$entity_type,
+        $entity_type_manager,
+        $messenger,
+        $logger_factory,
+      ])
+      ->setMethods(NULL)
+      ->getMock();
+
+    $this->userAuthenticator = $this->getMockBuilder(UserAuthenticator::class)
+      ->setConstructorArgs([$current_user,
+        $messenger,
+        $logger_factory,
+        $user_manager,
+        $this->dataHandler,
+      ])
+      ->setMethods(NULL)
+      ->getMock();
+
+  }
+
+  /**
+   * @covers Drupal\social_api\User\UserManager::setPluginId
+   */
+  public function testSetPluginId() {
+    $this->assertEquals(NULL, $this->userManager->getPluginId());
+    $this->userManager->setPluginId('social_auth_test');
+    $this->assertEquals('social_auth_test', $this->userManager->getPluginId());
+  }
+
+  /**
+   * @covers Drupal\social_api\User\UserManager::getPluginId
+   */
+  public function testGetPluginId() {
+    $this->userManager->setPluginId('social_auth_test2');
+    $this->assertEquals('social_auth_test2', $this->userManager->getPluginId());
+  }
+
+  /**
+   * @covers Drupal\social_api\User\UserAuthenticator::setPluginId
+   */
+  public function testSetPluginIdAuthenticator() {
+    $this->assertEquals(NULL, $this->userAuthenticator->getPluginId());
+    $this->userAuthenticator->setPluginId('social_auth_test');
+    $this->assertEquals('social_auth_test', $this->userAuthenticator->getPluginId());
+  }
+
+  /**
+   * @covers Drupal\social_api\User\UserAuthenticator::getPluginId
+   */
+  public function testGetPluginIdAuthenticator() {
+    $this->userAuthenticator->setPluginId('social_auth_test2');
+    $this->assertEquals('social_auth_test2', $this->userAuthenticator->getPluginId());
+  }
+
+  /**
+   * @covers Drupal\social_api\User\UserAuthenticator::getSessionKeys
+   */
+  public function testGetSessionKeys() {
+    $sample_session = ['h78323' => '78t2gq2g7q', 'pawdwadawd' => 'cbzhzxc'];
+
+    $this->userAuthenticator->setSessionKeysToNullify($sample_session);
+    $this->assertEquals($sample_session, $this->userAuthenticator->getSessionKeys());
+  }
+
+  /**
+   * @covers Drupal\social_api\User\UserAuthenticator::setSessionKeysToNullify
+   */
+  public function testSetSessionKeysToNullify() {
+    $sample_session = ['h78323' => '78t2gq2g7q', 'pawdwadawd' => 'cbzhzxc'];
+
+    $this->assertNotEquals($sample_session, $this->userAuthenticator->getSessionKeys());
+    $this->userAuthenticator->setSessionKeysToNullify($sample_session);
```

Comment:
I believe that only the keys of `$sample_session` should be passed to `setSessionKeysToNullify`. See the first comment about reverting this function